cassandra/cassandra-docker.sh

```diff
@@ -121,18 +133,49 @@ if [ -n "$HELP" ]; then
   echo "  --keystore_password_file=KEYSTORE_PASSWORD"
   echo "        a file containing only the keystore password"
   echo
-  echo "  --truststore_file=TRUSTSTORE_FILE_LOCATION"
+  echo "  --truststore_nodes_file=TRUSTSTORE_NODES_FILE_LOCATION"
   echo "        the path to where the truststore is located"
   echo
-  echo "  --truststore_password=TRUSTSTORE_PASSWORD"
+  echo "  --truststore_nodes_password=TRUSTSTORE_NODES_PASSWORD"
   echo "        the password to use for the truststore"
   echo
-  echo "  --truststore_password_file=TRUSTSTORE_PASSWORD"
+  echo "  --truststore_nodes_password_file=TRUSTSTORE_NODES_PASSWORD"
   echo "        a file containing only the truststore password"
   echo
+  echo "  --truststore_client_file=TRUSTSTORE_CLIENT_FILE_LOCATION"
+  echo "        the path to where the truststore is located"
+  echo
+  echo "  --truststore_client_password=TRUSTSTORE_CLIENT_PASSWORD"
+  echo "        the password to use for the truststore"
+  echo
+  echo "  --truststore_client_password_file=TRUSTSTORE_CLIENT_PASSWORD"
+  echo "        a file containing only the truststore password"
+  echo
+  echo "  --truststore_nodes_authorities=TRUSTSTORE_NODES_AUTHORITIES"
+  echo "        a file containing all certificate authorities to trust as peers"
+  echo
+  echo "  --truststore_client_authorities=TRUSTSTORE_CLIENT_AUTHORITIES"
+  echo "        a file containing all certificate authorities to trust as clients"
+  echo
   exit 0
 fi
 
+CASSANDRA_HOME=${CASSANDRA_HOME:-"/opt/apache-cassandra"}
+CASSANDRA_CONF=${CASSANDRA_CONF:-"${CASSANDRA_HOME}/conf"}
+CASSANDRA_CONF_FILE=${CASSANDRA_CONF_FILE:-"${CASSANDRA_CONF}/cassandra.yaml"}
+KEYSTORE_DIR=${KEYSTORE_DIR:-"${CASSANDRA_CONF}"}
+
+SERVICE_ALIAS=${SERVICE_ALIAS:-"cassandra"}
+SERVICE_CERT=${SERVICE_CERT:-"/hawkular-cassandra-certs/tls.crt"}
+SERVICE_CERT_KEY=${SERVICE_CERT_KEY:-"/hawkular-cassandra-certs/tls.key"}
```

Comment:
> Do you mean that SERVICE_CERT_KEY should be a parameter to the script, like, say, --truststore_client_authorities ?

Yep

> Not sure I get what you mean by "in a directory and file pattern like this": the cert secret is a volume mount, so, it's indeed a local file, as far as the script is concern.

What I mean is that we are hardcoding the file location, which means the secret has to be loaded at a specific directory and has to contain specifically named values.

There is nothing wrong with using specific file locations like this, I just normally find its obscures a bit what is happening and how things are being used. And it means we are more stuck with the secret structure we have now.